plugins/TagFix_BadValue.py

```diff
@@ -112,8 +112,23 @@ def check(self, data, tags, check_list_open):
         for k in keys:
             if not self.Values_open.match(tags[k]):
                 if k in self.exceptions_open:
-                    if tags[k] in self.exceptions_open[k]:
-                        # no error if in exception list
+                    isWhitelisted = tags[k] in self.exceptions_open[k]
+                    if not isWhitelisted and ";" in tags[k]:
+                        for val in tags[k].split(";"):
+                            if val in self.exceptions_open[k]:
+                                # Whitelisted value
+                                isWhitelisted = True
+                                continue
+                            elif self.Values_open.match(val):
+                                # Normal value
+                                continue
+                            else:
+                                # Bad value
+                                isWhitelisted = False
+                                break
+
+                    if isWhitelisted:
+                        # No error if in exception list
                         continue
                 err.append({"class": 3040, "subclass": stablehash64(k), "text": T_("Concerns tag: `{0}`", '='.join([k, tags[k]])) })
```

Comment:
Try to simplify, not tested code. We probably can simply more, as sub-if are just `and`.
```suggestion
                    if tags[k] not in self.exceptions_open[k] and ";" in tags[k]:
                        if any(map(lambda val: val not in self.exceptions_open[k] and not self.Values_open.match(val), tags[k].split(";")):
                            err.append({"class": 3040, "subclass": stablehash64(k), "text": T_("Concerns tag: `{0}`", '='.join([k, tags[k]])) })
```

Reply:
I could not directly use this code as it would not trigger an error in case there's no `;` in the code, but I've added a commit that serves a similar shortening of the code